pkg/volume/volume.go

```diff
@@ -0,0 +1,316 @@
+		klog.Errorf("volume id empty")
+		return response, fmt.Errorf("volume id empty")
+	}
+
+	err := v.hostAPI.FormatVolume(volumeID)
+	if err != nil {
+		klog.Errorf("failed FormatVolume %v", err)
+		return response, err
+	}
+	return response, nil
+}
+
+func (v *Volume) WriteVolumeCache(context context.Context, request *WriteVolumeCacheRequest) (*WriteVolumeCacheResponse, error) {
+	klog.V(2).Infof("WriteVolumeCache: Request: %+v", request)
+	response := &WriteVolumeCacheResponse{}
+
+	volumeID := request.VolumeId
+	if volumeID == "" {
+		klog.Errorf("volume id empty")
+		return response, fmt.Errorf("volume id empty")
+	}
+
+	err := v.hostAPI.WriteVolumeCache(volumeID)
+	if err != nil {
+		klog.Errorf("failed WriteVolumeCache %v", err)
+		return response, err
+	}
+	return response, nil
+}
+
+func (v *Volume) ResizeVolume(context context.Context, request *ResizeVolumeRequest) (*ResizeVolumeResponse, error) {
+	klog.V(2).Infof("ResizeVolume: Request: %+v", request)
+	response := &ResizeVolumeResponse{}
+
+	volumeID := request.VolumeId
+	if volumeID == "" {
+		klog.Errorf("volume id empty")
+		return response, fmt.Errorf("volume id empty")
+	}
+	sizeBytes := request.SizeBytes
+	// TODO : Validate size param
+
+	err := v.hostAPI.ResizeVolume(volumeID, sizeBytes)
+	if err != nil {
+		klog.Errorf("failed ResizeVolume %v", err)
+		return response, err
+	}
+	return response, nil
+}
+
+func (v *Volume) VolumeStats(context context.Context, request *VolumeStatsRequest) (*VolumeStatsResponse, error) {
+	getVolumeStatsRequest := &GetVolumeStatsRequest{
+		VolumeId: request.VolumeId,
+	}
+	getVolumeStatsResponse, err := v.GetVolumeStats(context, getVolumeStatsRequest)
+	if err != nil {
+		return nil, fmt.Errorf("Forward to GetVolumeStats failed, err=%+v", err)
+	}
+	volumeStatsResponse := &VolumeStatsResponse{
+		VolumeSize:     getVolumeStatsResponse.TotalBytes,
+		VolumeUsedSize: getVolumeStatsResponse.UsedBytes,
+	}
+	return volumeStatsResponse, nil
+}
+
+func (v *Volume) GetVolumeStats(context context.Context, request *GetVolumeStatsRequest) (*GetVolumeStatsResponse, error) {
+	klog.V(2).Infof("GetVolumeStats: Request: %+v", request)
+	volumeID := request.VolumeId
+	if volumeID == "" {
+		return nil, fmt.Errorf("volume id empty")
+	}
+
+	totalBytes, usedBytes, err := v.hostAPI.GetVolumeStats(volumeID)
+	if err != nil {
+		klog.Errorf("failed GetVolumeStats %v", err)
+		return nil, err
+	}
+
+	klog.V(2).Infof("VolumeStats: returned: Capacity %v Used %v", totalBytes, usedBytes)
+
+	response := &GetVolumeStatsResponse{
+		TotalBytes: totalBytes,
+		UsedBytes:  usedBytes,
+	}
+
+	return response, nil
+}
+
+func (v *Volume) GetVolumeDiskNumber(context context.Context, request *VolumeDiskNumberRequest) (*VolumeDiskNumberResponse, error) {
+	getDiskNumberFromVolumeIDRequest := &GetDiskNumberFromVolumeIDRequest{
+		VolumeId: request.VolumeId,
+	}
+	getDiskNumberFromVolumeIDResponse, err := v.GetDiskNumberFromVolumeID(context, getDiskNumberFromVolumeIDRequest)
```

Comment:
Give this is just a wrapper and around for API compat guarantees of csi-proxy, how about dropping `GetVolumeDiskNumber`?

Reply:
tracked in https://github.com/kubernetes-csi/csi-proxy/issues/241